security/self-assessment.md

```diff
@@ -0,0 +1,363 @@
+# Metal3 Self-assessment
+<!-- markdownlint-disable single-h1 no-emphasis-as-heading -->
+<!-- cSpell:ignore roadmapping,Kubed,controlplanes,htpasswd -->
+
+> This self-assessment document is intended to identify security insights of
+> Metal3 project, aid in roadmapping and a tool/documentation for onboarding new
+> maintainers to the project.
+
+Metal3 has been in CNCF Sandbox since 2020, and is now applying for
+[Incubation](https://github.com/cncf/toc/issues/1365).
+
+# Table of contents
+
+* [Metadata](#metadata)
+   * [Security links](#security-links)
+* [Overview](#overview)
+   * [Actors](#actors)
+   * [Actions](#actions)
+   * [Background](#background)
+   * [Goals](#goals)
+   * [Non-goals](#non-goals)
+* [Self-assessment use](#self-assessment-use)
+* [Security functions and features](#security-functions-and-features)
+* [Project compliance](#project-compliance)
+* [Secure development practices](#secure-development-practices)
+* [Security issue resolution](#security-issue-resolution)
+* [Appendix](#appendix)
+
+## Metadata
+
+|                   |                                              |
+| ----------------- | -------------------------------------------- |
+| Assessment Stage  | In Progress                                  |
+| Software          | <https://github.com/metal3-io/>              |
+| Security Provider | No                                           |
+| Languages         | Go, Bash, Python                             |
+| SBOM              | Metal3 does not generate SBOMs currently     |
+
+### Security links
+
+| Document      | URL                                              |
+| ------------- | ------------------------------------------------ |
+| Security file | <https://book.metal3.io/security_policy>         |
+
+## Overview
+
+The Metal3 Project (pronounced: “Metal Kubed”) empowers organizations with a
+flexible, open-source solution for bare metal provisioning that combines the
+benefits of bare metal performance with the ease of use and automation provided
+by Kubernetes. Metal3 provides comprehensive set of components for baremetal
+host management with Kubernetes so that a user can enroll user's baremetal
+machines, provision operating system images, deploy Kubernetes clusters on them
+and maintain the lifecycle of the kubernetes cluster, baremetal server and
+applications all through kubernetes native APIs.
```

Comment:
```suggestion
by Kubernetes. Metal3 provides a comprehensive set of components for baremetal
host management with Kubernetes, so that a user can enroll baremetal
machines, provision operating system images, deploy Kubernetes clusters on them
and manage the lifecycle of the Kubernetes cluster, baremetal server and
applications, all through Kubernetes native APIs.
```